demos/osal/lv_demo_osal.c

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file lv_demo_osal.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_demo_osal.h"
+
+#if LV_USE_DEMO_OSAL
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void counter_button_event_cb(lv_event_t * e);
+static void increment_thread_entry(void * user_data);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static lv_thread_sync_t press_sync;
+static lv_thread_t increment_thread;
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_demo_osal(void)
+{
+    lv_obj_t * counter_button;
+
+    counter_button = lv_button_create(lv_screen_active());
+    lv_obj_align(counter_button, LV_ALIGN_CENTER, 0, -15);
+    lv_obj_add_event_cb(counter_button, counter_button_event_cb, LV_EVENT_CLICKED, NULL);
+
+    if(lv_thread_sync_init(&press_sync) != LV_RESULT_OK) {
+        LV_LOG_ERROR("Error initializing thread sync");
+    }
+
+    if(lv_thread_init(&increment_thread, LV_THREAD_PRIO_MID, increment_thread_entry, 2048, NULL) != LV_RESULT_OK) {
+        LV_LOG_ERROR("Error initializing thread");
+    }
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+void counter_button_event_cb(lv_event_t * e)
+{
+    if(lv_thread_sync_signal(&press_sync) != LV_RESULT_OK) {
+        LV_LOG_ERROR("Error signaling thread sync");
+    }
+}
+
+void increment_thread_entry(void * user_data)
+{
+    lv_obj_t * counter_label;
+    uint32_t press_count = 0;
+
+    counter_label = lv_label_create(lv_scr_act());
+    lv_obj_align(counter_label, LV_ALIGN_CENTER, 0, 0);
```

Comment:
Don't we need a mutex here? What if this thread is running concurrently with the thread of `lv_timer_handler`?

Reply:
True, but this would mean that the `lv_timer_handler` thread would also need to take this lock. I do not understand how this can be achieved though.

Reply:
Correct. See the latest comments in #6059

Reply:
So if I followed the disussion correctly there will be some sort of global mutex that may be taken on the PR for that is ready, right?